baseapp/abci.go

```diff
@@ -249,6 +243,18 @@ func (app *BaseApp) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBloc
 	return res
 }
 
+// ProcessProposal implements the ability for the application to verify and/or modify transactions in a block proposal.
```

Comment:
I thought only `ProcessProposal` verifies transactions. How do verifications between the two methods differ?

Reply:
Updated the godoc. In short, `ProcessProposal` implementation is really up to the application. A the very least it must ensure all txs in the proposed block are valid. However, it may also actually _execute_ the entire lock as well. Doing so allows for various improvements such as gas/fee refunds and even performance improvements if the app does it in parallel to Tendermint's PreVote and PreCommit rounds.